lasagne/layers/recurrent.py

```diff
@@ -757,3 +762,306 @@ def step_masked(input_dot_W_n, mask_n, cell_previous, hid_previous,
+            self.nonlinearity_resetgate = nonlinearities.identity
+        else:
+            self.nonlinearity_resetgate = nonlinearity_resetgate
+
+        if nonlinearity_updategate is None:
+            self.nonlinearity_updategate = nonlinearities.identity
+        else:
+            self.nonlinearity_updategate = nonlinearity_updategate
+
+        if nonlinearity_cell is None:
+            self.nonlinearity_cell = nonlinearities.identity
+        else:
+            self.nonlinearity_cell = nonlinearity_cell
+
+        self.learn_init = learn_init
+        self.num_units = num_units
+        self.grad_clipping = grad_clipping
+        self.backwards = backwards
+        self.gradient_steps = gradient_steps
+        self.unroll_scan = unroll_scan
+
+        # Input dimensionality is the output dimensionality of the input layer
+        num_batch = self.input_shape[0]
+        num_inputs = np.prod(self.input_shape[2:])
+        self.num_inputs = num_inputs
+
+        self.W_in_to_updategate = self.add_param(
+            W_in_to_updategate, (num_inputs, num_units),
+            name="W_in_to_updategate")
+
+        self.W_hid_to_updategate = self.add_param(
+            W_hid_to_updategate, (num_units, num_units),
+            name="W_hid_to_updategate")
+
+        self.b_updategate = self.add_param(
+            b_updategate, (num_units,),
+            name="b_updategate", regularizable=False)
+
+        self.W_in_to_resetgate = self.add_param(
+            W_in_to_resetgate, (num_inputs, num_units),
+            name="W_in_to_resetgate")
+
+        self.W_hid_to_resetgate = self.add_param(
+            W_hid_to_resetgate, (num_units, num_units),
+            name="W_hid_to_resetgate")
+
+        self.b_resetgate = self.add_param(
+            b_resetgate, (num_units,),
+            name="b_resetgate", regularizable=False)
+
+        self.W_in_to_cell = self.add_param(
+            W_in_to_cell, (num_inputs, num_units), name="W_in_to_cell")
+
+        self.W_hid_to_cell = self.add_param(
+            W_hid_to_cell, (num_units, num_units), name="W_hid_to_cell")
+
+        self.b_cell = self.add_param(
+            b_cell, (num_units,), name="b_cell", regularizable=False)
+
+        self.W_in_to_gates = T.concatenate(
+            [self.W_in_to_resetgate, self.W_in_to_updategate,
+             self.W_in_to_cell], axis=1)
+
+        # Same for hidden to gate weight matrices
+        self.W_hid_to_gates = T.concatenate(
+            [self.W_hid_to_resetgate, self.W_hid_to_updategate,
+             self.W_hid_to_cell], axis=1)
+
+        # Stack gate biases into a (4*num_units) vector
+        self.b_gates = T.concatenate(
+            [self.b_resetgate, self.b_updategate,
+             self.b_cell], axis=0)
+
+        # Initialize hidden state
+        if isinstance(hid_init, T.TensorVariable):
+            if hid_init.ndim != 2:
+                raise ValueError("When a tensor hid_init should be a matrix")
```

Comment:
For all of these value errors, "tensor" should be changed to "TensorVariable" (a matrix is always a tensor ;) )  Needs to be changed in all layers.